metron-interface/metron-alerts/src/app/pcap/pcap-filters/pcap-filters.component.ts

```diff
@@ -15,63 +15,116 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-import {Component, Input, Output, EventEmitter, OnInit, OnChanges, SimpleChanges} from '@angular/core';
+import {Component, Input, Output, EventEmitter, OnChanges, SimpleChanges} from '@angular/core';
+import { FormGroup, FormControl, Validators, ValidationErrors } from '@angular/forms';
+
 import * as moment from 'moment/moment';
 import { DEFAULT_TIMESTAMP_FORMAT } from '../../utils/constants';
 
 import { PcapRequest } from '../model/pcap.request';
 
+const DEFAULT_END_TIME = new Date();
```

Comment:
I noticed these static values and functions are outside of the PcapFiltersComponent class. What are the benefits to doing things this way instead of keeping them inside the PcapFiltersComponent class? Or if they are intended to be shared, why don't we move them into the utils folder and import as needed?

Reply:
For the record, in this case, it would be ok to put these things inside the class definition because they're connected to the pcap filter so they would be hardly reused by other components, but who knows. It's hard to predict.

I'm not sure about how these kind of things work in Java or other OOP heavy programming languages but Javascript (NodeJS) doesn't enforce us to put constants and static functions inside a class. 

The functions are static pure functions. They have nothing to do with the class' instance (this) so they should not be a private or a public class methods. they could be a static functions (a property of the class object itself). From testing perspective, I would keep it as is but export them of course so we wouldn't have to create a new instance of the component to test them. The best would be if they were replaced into a separate file in the appropriate folder based on whether it's shared across other components.

Reply:
These names do look like they are likely to collide with other potential areas of functionality though, so if they are genuinely global concerns, then fine, otherwise they should probably belong to some form of scope related to the relevant component.

Reply:
They belong to the file's scope.

They could collide with others but we can import them in this way:

```
import { DEFAULT_END_TIME as DEFAULT_PCAP_FILTER_END_TIME } from './pcap-filter.component';
```

But I can hardly imagine that we would use them somewhere else out of this scope.

Reply:
I think the DEFAULT_START_TIME and DEFAULT_END_TIME as all uppercase variables are a bit miss leading. These are not static finals like in java. These are actually private constants. I suggest to rename them to normal camelCase. DEFAULT_END_TIME might be reusable somewhere else but DEFAULT_START_TIME in its current form really belongs to the filtering. I think the same true to the validator and the transform functions too. Validator might be reusable if we turn our other forms reactive too in the future. For me, the question is how much we like to prepare our current code to an imagined future. And how much it supports the understanding of the existing functionality. I would leave it as it is right now but opened to refactor it if we see value in it.

Reply:
Agreed with @tiborm on renaming those constants. In fact, the Angular Style Guide [prefers to use lower camel case (though it also suggests tolerating existing upper snake case) for constants.](https://angular.io/guide/styleguide#constants) I also prefer not to worry about an imagined future, but I do think we should move some of this out to separate files if it doesn't belong in the class to make the file more readable, if anything.

Reply:
@sardell @ruffle1986  I'll refactor out the whole PcapRequest related code from the filter. It was part of the original implementation but I think it does not belongs to here. Filter component should produce a filter output instead of a request object.